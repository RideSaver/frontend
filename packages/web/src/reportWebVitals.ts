import type { CLSMetric, Metric, FCPMetric } from "web-vitals";

export default async function reportWebVitals(
    onPerfEntry?: (report: CLSMetric | Metric | FCPMetric) => void
) {
    if (onPerfEntry && onPerfEntry instanceof Function) {
        const { onCLS, onFID, onINP, onFCP, onLCP, onTTFB } = await import(
            "web-vitals"
        );
        console.log(onCLS);
        onCLS(onPerfEntry);
        onFID(onPerfEntry);
        onINP(onPerfEntry);
        onFCP(onPerfEntry);
        onLCP(onPerfEntry);
        onTTFB(onPerfEntry);
    }
}
