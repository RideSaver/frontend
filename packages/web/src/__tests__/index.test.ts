import { describe, test } from "@jest/globals";
import "@testing-library/jest-dom";

import reportWebVitals from "../reportWebVitals";
import MainElement from "../App";
import reactDOM from "react-dom/client";

jest.mock("../App");
jest.mock("../reportWebVitals", () => (jest.fn()));
console.log(reportWebVitals);

const renderSpy = jest.spyOn(reactDOM, "createRoot");
const hydrateSpy = jest.spyOn(reactDOM, "hydrateRoot");

function setupRoot(innerHTML: string = "") {
    document.body.innerHTML = `<div id="root">${innerHTML}</div>`;
    // eslint-disable-next-line testing-library/no-node-access
    return document.getElementById("root");
}

describe("Web Entrypoint", () => {
    test("Calls web vitals reporter", async () => {
        setupRoot();
        await import("..");
        expect(reportWebVitals).toHaveBeenCalled();
    });
    test("Renders app if there is no HTML", async () => {
        const root = setupRoot();
        await import("..");
        expect(renderSpy).toHaveBeenCalledWith(MainElement, root);
        expect(hydrateSpy).not.toHaveBeenCalled();
        expect(root).toMatchSnapshot();
    });
    test("Hydrates app if there is HTML from SSR", async () => {
        const root = setupRoot("<Main />");
        await import("..");
        expect(renderSpy).not.toHaveBeenCalled();
        expect(hydrateSpy).toHaveBeenCalledWith(MainElement, root);
    });
});
