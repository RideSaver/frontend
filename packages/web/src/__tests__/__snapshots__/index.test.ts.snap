// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Themes Calls web vitals reporter 1`] = `
Document {
  "location": Location {
    "assign": [Function],
    "hash": "",
    "host": "localhost",
    "hostname": "localhost",
    "href": "http://localhost/",
    "origin": "http://localhost",
    "pathname": "/",
    "port": "",
    "protocol": "http:",
    "reload": [Function],
    "replace": [Function],
    "search": "",
    "toString": [Function],
  },
}
`;
